source/merge-deep.d.ts

```diff
@@ -0,0 +1,469 @@
+	? DefaultType
+	: Destination extends UnknownRecord
+		? Source extends UnknownRecord
+			? MergeDeepRecord<Destination, Source, Options>
+			: DefaultType
+		: Destination extends UnknownArrayOrTuple
+			? Source extends UnknownArrayOrTuple
+				? MergeDeepArrayOrTuple<Destination, Source, Merge<Options, {spreadTopLevelArrays: false}>>
+				: DefaultType
+			: DefaultType>;
+
+/**
+MergeDeep options.
+
+@see {@link MergeDeep}
+*/
+export type MergeDeepOptions = {
+	/**
+	Merge mode for array and tuple.
+
+	When we walk through the properties of the objects and the same key is found and both are array or tuple, a merge mode must be chosen:
+	- `spread`: Spreads the destination and the source values. This is the default mode.
+	- `replace`: Replaces the destination value by the source value.
+
+	See {@link MergeDeep} for usages and examples.
+
+	@default 'spread'
+	*/
+	arrayMergeMode?: ArrayMergeMode;
+
+	/**
+	Whether to affect the individual elements of arrays and tuples.
+
+	If this option is set to `true` the same merge rules as for object properties are applied:
+
+	- Elements that only exist in one array are copied into the new array.
+	- Elements that exist in both arrays are merged if possible or replaced by the one of the source if not.
+	- By default, top-level arrays and tuples are spread. See {@link MergeDeepOptions.arrayMergeMode arrayMergeMode} option to change this behaviour.
+
+	@default false
+	*/
+	recurseIntoArrays?: boolean;
+};
+
+/**
+Internal options.
+*/
+type MergeDeepInternalOptions = Merge<MergeDeepOptions, {spreadTopLevelArrays?: boolean}>;
+
+/**
+Merge default and internal options with user provided options.
+*/
+type DefaultMergeDeepOptions<Options extends MergeDeepOptions> = Merge<{
+	arrayMergeMode: 'replace';
+	recurseIntoArrays: false;
+	spreadTopLevelArrays: true;
+}, Options>;
+
+/**
+This utility selects the correct entry point with the corresponding default options. This avoids re-merging the options at each iteration.
+*/
+type MergeDeepWithDefaultOptions<Destination, Source, Options extends MergeDeepOptions> = SimplifyDeep<
+[undefined] extends [Destination | Source]
+	? never
+	: Destination extends UnknownRecord
+		? Source extends UnknownRecord
+			? MergeDeepRecord<Destination, Source, DefaultMergeDeepOptions<Options>>
+			: never
+		: Destination extends UnknownArrayOrTuple
+			? Source extends UnknownArrayOrTuple
+				? MergeDeepArrayOrTuple<Destination, Source, DefaultMergeDeepOptions<Options>>
+				: never
+			: never
+>;
+
+/**
+Merge two objects or two arrays/tuples recursively into a new type.
+
+- Properties that only exist in one object are copied into the new object.
+- Properties that exist in both objects are merged if possible or replaced by the one of the source if not.
+- Top-level arrays and tuples are always spread.
+- By default, inner arrays and tuples are replaced. See {@link MergeDeepOptions.arrayMergeMode arrayMergeMode} option to change this behaviour.
```

Comment:
^ Here it says `replaced`. But here it says `spread`: https://github.com/sindresorhus/type-fest/pull/452/files#diff-5ce63d84bd0497ef6dd72ef89147e7bf80f464a1185b1a34139c6c1cb6c37ee5R307